### PR TITLE
Add heartbeat support for V2

### PR DESCRIPTION
**Pull request by fearful-symmetry**
## What does this PR do?

This PR adds V2 support for heartbeat running under elastic-agent.

I haven't thoroughly tested this yet, but I wanted to put in the PR while I juggle the 2-3 other things I'm working on. See the Auditbeat PR for more detailed instructions on how to test: https://github.com/elastic/beats/pull/33026

Here's an example of an elastic-agent input config that should work with heartbeat:
```
- id: synthetics/http-synthetics-f42c9747-2455-491b-a66c-967adc98007a
    name: test
    revision: 1
    type: synthetics/http
    use_output: default
    meta:
      package:
        name: synthetics
        version: 0.10.2
    data_stream:
      namespace: default
    package_policy_id: f42c9747-2455-491b-a66c-967adc98007a
    streams:
      - id: synthetics/http-http-f42c9747-2455-491b-a66c-967adc98007a
        name: test
        type: http
        enabled: true
        data_stream:
          dataset: http
          type: synthetics
        urls: 'http://localhost:9200'
        schedule: '@every 3m'
        timeout: 16s
        max_redirects: 0
        response.include_headers: true
        response.include_body: on_error
        check.request.method: GET
        processors:
          - add_observer_metadata:
              geo:
                name: Fleet managed
          - add_fields:
              target: ''
              fields:
                monitor.fleet_managed: true
```

## Checklist

- [x] Test under agent 
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

### CODE REVIEW

**Comment by fearful-symmetry** [comment]
Alright, finally managed to test this.

**Comment by paulb-elastic** [comment]
Pinging @emilioalvap and @andrewvc as there was some discussion about this regarding https://github.com/elastic/beats/issues/32163

**Comment by andrewvc** [comment]
@fearful-symmetry with the V2 protocol does the `output.elasticsearch` section still get written? https://github.com/elastic/beats/issues/32163 , which was merged recently, depends on it, reading back from ES to get the last known state for heartbeat

**Comment by andrewvc** [comment]
You can see where we set this up, and how, here: https://github.com/elastic/beats/blob/main/heartbeat/beater/heartbeat.go#L72

**Comment by fearful-symmetry** [comment]
@andrewvc 

> with the V2 protocol does the output.elasticsearch section still get written?

Yes, it does. Until the shipper comes along, we're still using built-in elasticsearch output. Theoretically, nothing should change in terms of the actual configs that are sent to individual components of a beat; this V2 support mechanism is just a compatibility shim for V2.

**Comment by fearful-symmetry** [comment]
> You can see where we set this up, and how, here: https://github.com/elastic/beats/blob/main/heartbeat/beater/heartbeat.go#L72

@andrewvc does this work under the current fleet setup? This happens outside of the reloaded registers here: https://github.com/elastic/beats/blob/main/heartbeat/beater/heartbeat.go#L213, which means that the code you linked to (I think) won't respond to changes in the fleet config, regardless of this PR.

**Comment by andrewvc** [comment]
@fearful-symmetry I'm realizing it's only been tested on the beat, thanks a bunch for noticing that, I've opened https://github.com/elastic/beats/issues/33244 to track.

File: x-pack/heartbeat/cmd/root.go [hunk 1]
```diff
@@ -5,16 +5,39 @@
...
+	"github.com/elastic/elastic-agent-client/v7/pkg/proto"
 
 	_ "github.com/elastic/beats/v7/x-pack/libbeat/include"
+	"github.com/elastic/beats/v7/x-pack/libbeat/management"
 )
 
 // RootCmd to handle beats cli
 var RootCmd *cmd.BeatsRootCmd
 
+// heartbeatCfg is a callback registered via SetTransform that returns a packetbeat Elastic Agent client.Unit
```
Comment (blakerouse) [inline x-pack/heartbeat/cmd/root.go:23]: Is the "packetbeat" comment correct there? Seems it should be "heartbeat".

**Review by blakerouse — APPROVED** [review]
Looks good to me, minus the one inline comment.

**Pull request merged**